controllers/atlasmigration_controller.go

```diff
@@ -173,12 +184,67 @@ func (r *AtlasMigrationReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		r.recordErrEvent(res, err)
 		return result(err)
 	}
+	if data.Dir != nil {
+		// Compress the migration directory then store it in the secret
+		// for later use when atlas runs the migration down.
+		if err := r.storeMigrations(ctx, res, data.Dir); err != nil {
+			res.SetNotReady("StoringMigrationDir", err.Error())
+			r.recordErrEvent(res, err)
+			return result(err)
+		}
+	}
 	status.ObservedHash = hash
 	res.SetReady(*status)
 	r.recorder.Eventf(res, corev1.EventTypeNormal, "Applied", "Version %s applied", status.LastAppliedVersion)
 	return ctrl.Result{}, nil
 }
 
+func (r *AtlasMigrationReconciler) readMigrations(ctx context.Context, res *dbv1alpha1.AtlasMigration) (migrate.Dir, error) {
```

Comment:
migrations is a pretty overloaded term in this codebase. 
maybe `dirState`, e.g `readDirState`, `storeDirState` etc.?

Reply:
Updated;